app/TestKernel.php

```diff
@@ -42,14 +43,27 @@ protected function cleanupContainer($container)
             }
 
             $serviceObject = new \ReflectionObject($service);
-            foreach ($serviceObject->getProperties() as $prop) {
-                $prop->setAccessible(true);
-                if ($prop->isStatic()) {
-                    continue;
+
+            $properties = $serviceObject->getProperties();
+            // If it has a static property it could be a singleton, therefore we should not
+            // reset it's properties
+            $isStatic = false;
+            foreach ($properties as $property) {
+                if ($property->isStatic()) {
+                    $isStatic = true;
```

Comment:
That's a little bit of sorcery, but we can remove `$isStatic` and put `continue 2;` here and it will jump to the beginning of the next iteration of services loop :)